src/SpectatorView.Unity/Assets/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodeMarkerDetector.cs

```diff
@@ -88,26 +93,43 @@ public bool TryGetMarkerSize(int markerId, out float size)
         }
 
 #if ENABLE_QRCODES
-        protected void Start()
+        protected async void OnEnable()
         {
             if (_qrCodesManager == null)
             {
                 _qrCodesManager = QRCodesManager.FindOrCreateQRCodesManager(gameObject);
+            }
+
+            if (_qrCodesManager != null)
+            {
                 _qrCodesManager.QRCodeAdded += QRCodeAdded;
                 _qrCodesManager.QRCodeRemoved += QRCodeRemoved;
                 _qrCodesManager.QRCodeUpdated += QRCodeUpdated;
+                await StartTracking();
             }
+        }
 
-            StartTracking();
+        protected void OnDisable()
+        {
+            if (_qrCodesManager != null)
+            {
+                StopTracking();
+                _qrCodesManager.QRCodeAdded -= QRCodeAdded;
+                _qrCodesManager.QRCodeRemoved -= QRCodeRemoved;
+                _qrCodesManager.QRCodeUpdated -= QRCodeUpdated;
+            }
         }
 
-        private async void StartTracking()
+        private async Task StartTracking()

```

Comment:
>StartTracking [](start = 27, length = 13)

Nit: I prefer the naming convention where async methods have a name that ends in Async (StartTrackingAsync). Makes it a lot easier when reading code to tell if the method is synchronous or not